.gitignore

```diff
@@ -1,3 +1,5 @@
+pubspec.lock
```

Comment:
Why ignore pubspec.lock?

Reply:
So that application packages have a little bit more freedom in version solving. https://dart.dev/guides/libraries/private-files

I'm happy to pull it out if you'd like to keep the lock.

Reply:
Ah it's fine

Reply:
To be honest, I've never had it be an issue either way. But I've been on teams where the rule is followed, so it's one of those things I do.

Reply:
Ah, and what I said about freedom in solving isn't correct. Looks like it's recommended so that packages are always up-to-date for testing in CI? TIL. :)

Reply:
Sounds like a weird argument to me. I would expect the opposite reasoning, where we want the entire team and the CI to use the same versions

But i don't really care 😊